src/HealthChecks/HealthChecks/src/DefaultHealthCheckService.cs

```diff
@@ -46,7 +46,7 @@ internal class DefaultHealthCheckService : HealthCheckService
             }
 
             var totalTime = ValueStopwatch.StartNew();
-            Log.HealthCheckProcessingBegin(_logger);
+            HealthCheckProcessingBegin();
```

Comment:
Stupid q - how does the implementation know to use the `_logger` instance?

Reply:
When generating instance-mode logging methods, the field that holds the logger is determined by looking at available fields. You get an error if there isn't a field of the right type or if there is more than one such field.

Refer to: [logging-generator.md](https://github.com/dotnet/designs/blob/main/accepted/2021/logging-generator.md#sample-with-log-instance-method) for more info.